### ISSUE TITLE
Embed option for getResources on API Gateway

### ISSUE THREAD

**Issue by southpolesteve** [aws/aws-sdk-js#764]
https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/APIGateway.html#getRestApis-property

The docs specify the existence of an `embed` option but no more details about how to use it. I have used it directly via the rest api, but can't seem to get it to work in the SDK. When adding `embed: true` or `embed: "methods"` the to the `params`object the SDK throws an invalid params error:

```
{ [UnexpectedParameter: Unexpected key 'embed' found in params]
  message: 'Unexpected key \'embed\' found in params',
  code: 'UnexpectedParameter',
  time: Mon Oct 26 2015 23:24:25 GMT-0400 (EDT) }
```

**Comment by trevorrowe** [comment]
@southpolesteve I've attempted to make the API call to get a resource while passing the querystring `embed=true` and I haven't seen anything different come across the wire in the response. Can you give an example of what you get back when using the API directly?

**Comment by southpolesteve** [comment]
@trevorrowe The docs for this directly in the rest api are here: http://docs.aws.amazon.com/apigateway/api-reference/link-relation/restapi-resources/

I believe you have to set embed=methods in the query string. I will add an example later this afternoon

**Comment by trevorrowe** [comment]
@southpolesteve Thanks for the tip. It looks like this can be addressed by the SDK by adding the ability to specify a list of resources to embed via the querystring. Using `embed=methods` I was able to produce what you are expecting. Some minor SDK updates are required for this to work.

**Comment by southpolesteve** [comment]
@trevorrowe @chrisradek I've just submitted a PR for this: https://github.com/aws/aws-sdk-js/pull/869

**Comment by southpolesteve** [comment]
@chrisradek any updates? I've got a downstream library waiting for this change. Currently running on my now out of date patched version.

**Comment by chrisradek** [comment]
@southpolesteve 
Sorry for the delay. The changes required are to the models that all the SDKs share and go a little beyond what's in the pr. Not all the SDKs support the functionality that is required for this change, but we are pushing for them to make the changes necessary to support this. 

I know how frustrating it is to hear "it's coming soon" but I talked with the SDKs that need to update to support this and they have it as something to take care of in the near future.

**Comment by southpolesteve** [comment]
No worries. Thanks for the update!

**Comment by chrisradek** [comment]
@southpolesteve 
In the meantime, I posted a workaround in the related issue #941.

You can add the `embed=methods` to the query string of your request's path manually:
```javascript
var req = apigateway.getResources({
    restApiId: 'apiId'
});

req.on('build', function(req) {
    req.httpRequest.path += '/?embed=methods';
});

req.send(function(err, data) {
    if (err) {
        console.log(err);
    } else {
        console.log(data);
    }
});
```

This at least can help get around having to use a patched version of the SDK, but would have to be taken out once the SDK is updated with the `embed` fix.

**Comment by dankantor** [comment]
@chrisradek when I try that workaround I get this error:

`InvalidQueryStringException: null`

**Comment by dankantor** [comment]
Ok just figured out my bug, I already had `limit` set so just needed to add it this way `req.httpRequest.path += '&embed=methods';`

**Comment by chrisradek** [comment]
Closing since `embed` is available in `v2.43.0` of the SDK. Please let me know if you still encounter issues.
